suppressible-error-prone/src/test/java/com/palantir/suppressibleerrorprone/SuppressWarningsUtilsTest.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * (c) Copyright 2025 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.suppressibleerrorprone;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.List;
+import java.util.Set;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+class SuppressWarningsUtilsTest {
+    @Nested
+    class ModifySuppressions {
+        @Test
+        void no_human_authored_suppressions() {
+            assertThat(SuppressWarningsUtils.modifySuppressions(List.of(), Set.of("New")))
+                    .containsExactly("for-rollout:New");
+        }
+
+        @Test
+        void automatic_suppresions_go_after_human_authored_suppressions() {
+            assertThat(SuppressWarningsUtils.modifySuppressions(List.of("Something"), Set.of("New")))
+                    .containsExactly("Something", "for-rollout:New");
+        }
+
+        @Test
+        void maintains_human_authored_ordering() {
+            assertThat(SuppressWarningsUtils.modifySuppressions(List.of("C", "A", "B"), Set.of("ArrayToString")))
+                    .containsExactly("C", "A", "B", "for-rollout:ArrayToString");
+        }
+
+        @Test
+        void auto_suppressions_are_initially_alphabetically_ordered() {
+            assertThat(SuppressWarningsUtils.modifySuppressions(List.of("A"), Set.of("ArrayEquals", "ArrayToString")))
+                    .containsExactly("A", "for-rollout:ArrayEquals", "for-rollout:ArrayToString");
+        }
+
+        @Test
+        void maintains_alphabetical_order_for_automated_suppressions() {
+            assertThat(SuppressWarningsUtils.modifySuppressions(
+                            List.of("Blah", "for-rollout:A", "for-rollout:Something"), Set.of("ArrayToString")))
+                    .containsExactly("Blah", "for-rollout:A", "for-rollout:ArrayToString", "for-rollout:Something");
+        }
+
+        @Test
+        void reorders_automated_suppresions_to_the_end() {
```

Comment:
I think you want another test where it's:

``java
assertThat(SuppressWarningsUtils.modifySuppressions(
                            List.of("Blah", "for-rollout:A", "HumanAuthored"), Set.of("ArrayToString")))
```

where you end up with `"Blah", "HumanAuthored", "for-rollout:A", "for-rollout:ArrayToString"`

this is for the case where someone has added another suppression after some started rolling out